packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -616,8 +616,42 @@ const PostsPage = ({fullPost, postPreload, eagerPostComments, refetch, classes}:
 
   const marketInfo = getMarketInfo(post)
 
+  // check for deep equality between terms and eagerPostComments.terms
+  const useEagerResults = eagerPostComments && isEqual(commentTerms, eagerPostComments?.terms);
+
+  const lazyResults = useMulti({
+    terms: {...commentTerms, postId: post._id},
+    skip: useEagerResults,
+    ...postsCommentsThreadMultiOptions,
+  });
+
+  const { loading, results, loadMore, loadingMore, totalCount } = useEagerResults ? eagerPostComments.queryResponse : lazyResults;
```

Comment:
Moved from below because we now need to check whether the hash comment exists in the page